azkaban-common/src/main/java/azkaban/project/JdbcProjectImpl.java

```diff
@@ -1015,31 +1015,37 @@ public void uploadFlowFile(final int projectId, final int projectVersion, final
 
   @Override
   public File getUploadedFlowFile(final int projectId, final int projectVersion,
-      final int flowVersion, final String flowName) throws ProjectManagerException {
+      final String flowFileName, final int flowVersion, final File tempDir)
```

Comment:
why is tempDir an argument for this method?

Reply:
This is to delete the tempDir after the returned flow file is no longer used. We cannot do the deletion inside this method because we still need to reference the flow file in the tempDir.